apps/studio/src/components/ResourceSelector/ResourceSelectorContent.tsx

```diff
@@ -0,0 +1,150 @@
+import { Text, VStack } from "@chakra-ui/react"
+import { Button } from "@opengovsg/design-system-react"
+
+import type { ResourceItemContent } from "~/schemas/resource"
+import { ResourceItem, ResourceItemSkeleton } from "./ResourceItem"
+import { lastResourceItemInAncestryStack } from "./utils"
+
+const NoItemsInFolderResult = () => {
+  return (
+    <Text
+      h="full"
+      w="full"
+      textAlign="center"
+      textStyle="caption-2"
+      display="flex"
+      alignItems="center"
+      justifyContent="center"
+    >
+      This folder is empty.
+    </Text>
+  )
+}
+
+const ResourceItemsResults = ({
+  resourceItemsWithAncestryStack,
+  isResourceIdHighlighted,
+  isResourceItemDisabled,
+  hasAdditionalLeftPadding,
+  handleClickResourceItem,
+}: {
+  resourceItemsWithAncestryStack: ResourceItemContent[][]
+  isResourceIdHighlighted: (resourceId: string) => boolean
+  isResourceItemDisabled: (resourceItem: ResourceItemContent) => boolean
+  hasAdditionalLeftPadding: boolean
+  handleClickResourceItem: (
+    resourceItemWithAncestryStack: ResourceItemContent[],
+  ) => void
+}) => {
+  return resourceItemsWithAncestryStack.map((resourceItemWithAncestryStack) => {
+    const lastChild = lastResourceItemInAncestryStack(
+      resourceItemWithAncestryStack,
+    )
+
+    if (!lastChild) {
+      throw new Error(
+        "Unexpected undefined lastChild from lastResourceItemInAncestryStack",
+      )
+    }
```

Comment:
double check - usually when we throw errors, we should have a `ErrorBoundary` to recover. this allows us to 
1. isolate the error cleanly 
2. allow the users to self-recover, if possible. 

i think not a big deal if htis error is indeed never expected (and hence crashworthy) as we have top level error boundaries for precisely this situation (unrecoverable errors)